builtin/providers/vsphere/resource_vsphere_virtual_machine_test.go

```diff
@@ -33,7 +33,7 @@ func TestAccVSphereVirtualMachine_basic(t *testing.T) {
 	if v := os.Getenv("VSPHERE_DATASTORE"); v != "" {
 		datastoreOpt = fmt.Sprintf("        datastore = \"%s\"\n", v)
 	}
-	template := os.Getenv("VSPHERE_TEMPLATE")
+	cloneSource := os.Getenv("VSPHERE_CLONE_SOURCE")
```

Comment:
Yep I know. It would be nice to refactor the vsphere provider module from A to Z. Use proper structuring, etc.

When things are "not so good" all over the place, people (me included) tend to continue that pattern. No excuse, but it's a fact it happens.